rskj-core/src/main/java/co/rsk/rpc/ExecutionBlockRetriever.java

```diff
@@ -126,4 +127,30 @@ private static BlockResult newBlockResult(Block block) {
                 null
         );
     }
+
+    @Override
+    public void start() {
+        emitter.addListener(listener);
+    }
+
+    @Override
+    public void stop() {
+        emitter.removeListener(listener);
+    }
+
+    private class CachedResultCleaner extends EthereumListenerAdapter {
+        @Override
+        public void onPendingTransactionsReceived(List<Transaction> transactions) {
+            cleanCachedResult();
```

Comment:
What happens if `transactions` is empty because received transaction was not valid? Should we be clearing the cache in this case?

Reply:
Ok, just saw that is not possible, the event is not emitted in that case ([here](https://github.com/rsksmart/rskj/blob/995ac70e109b164b63992dd7d253c970da36fdce/rskj-core/src/main/java/co/rsk/core/bc/TransactionPoolImpl.java#L189)) 👍